src/crypto/hpke/mod.rs

```diff
@@ -0,0 +1,236 @@
+//! Provides decryption primitives for HPKE according to the [`specification`].
+//!
+//! [`specification`]: https://github.com/patcg-individual-drafts/ipa/pull/31
+
+use hpke::aead::AeadTag;
+use hpke::{single_shot_open_in_place_detached, Deserializable, OpModeR};
+use std::{fmt, io};
+
+mod aad;
+mod registry;
+
+pub use aad::AssociatedData;
+pub use registry::KeyRegistry;
+
+/// IPA ciphersuite
+type IpaKem = hpke::kem::X25519HkdfSha256;
+type IpaAead = hpke::aead::AesGcm128;
+type IpaKdf = hpke::kdf::HkdfSha256;
+
+pub type KeyIdentifier = u8;
+pub type Epoch = u16;
+type IpaPublicKey = <IpaKem as hpke::kem::Kem>::PublicKey;
+type IpaPrivateKey = <IpaKem as hpke::kem::Kem>::PrivateKey;
+
+#[derive(Debug)]
+pub struct BottomError;
```

Comment:
Bottom?  DecryptionError or DecapsulationError perhaps.

Reply:
yep that is what I meant here